### ISSUE TITLE
Did you sabotage other projects?

### ISSUE THREAD

**Issue by jberkus** [QiushiWu/qiushiwu.github.io#5]
Speaking as a maintainer on other OSS projects, I think we all need to know if you sabotaged anything else.

What other OSS projects did you "experiment" with, other than the Linux Kernel?

**Comment by ABotond** [comment]
At least according to his paper he did not even sabotage the Linux Kernel.
He has sent in the patches per e-mail and after they were approved he immediately notified the maintainers and provided them the correct fixes. None of his vulnerabilities became even just a git commit.

**Comment by IThinkImOKAY** [comment]
I mean, what in the actual fuck? How ignorant do you have to be to attempt to sabotage a kernel that over half the world relies on.

**Comment by sigaloid** [comment]
@ABotond https://lore.kernel.org/lkml/78ac6ee8-8e7c-bd4c-a3a7-5a90c7ccb399@roeck-us.net/

**Comment by ABotond** [comment]
@sigaloid 
> @ABotond https://lore.kernel.org/lkml/78ac6ee8-8e7c-bd4c-a3a7-5a90c7ccb399@roeck-us.net/
What does this prove?
This is not one of the vulnerabilities from the paper. (According to the writeup they were not even submitted from an *@umn.edu address.) Guenter Roeck even points out he doesn't see a problem with this commit (besides its limited value). UMN made actually quite a few thousand useful contributions to the kernel.

What I actually don't get around this whole scandal is that why everybody is thinking that all the other Linux contributors are trustworthy.
The same could have been (already) done by the NSA/Chinese gov/Russian gov/etc long ago.

The paper gives a quite nice overview of the vulnerabilities introduced with small patches in the past years, how they worked and why they remained undetected for a long time.
It helps to raise awareness of the problem. I see it no different from finding a way to get in a pretty important health care system without causing any harm then reporting it.
OF COURSE it comes with an extra work for the maintainers - just as in case of any vulnerabilities in any systems.
Also, I find it good if the maintainers will be more suspicious in the future about the small patches. In the middle of an ongoing cyber cold war it is more than important.

**Comment by sigaloid** [comment]
>It helps to raise awareness of the problem. I see it no different from finding a way to get in a pretty important health care system without causing any harm then reporting it.

Without causing any harm? What if this got into an RC or god forbid a full-fledged release? They didn't notify upon commit, what makes you think they'd stop upon release?

**Comment by ABotond** [comment]
> > It helps to raise awareness of the problem. I see it no different from finding a way to get in a pretty important health care system without causing any harm then reporting it.
> 
> Without causing any harm? What if this got into an RC or god forbid a full-fledged release? They didn't notify upon commit, what makes you think they'd stop upon release?
> 
> https://lore.kernel.org/linux-nfs/YH%2F8jcoC1ffuksrf@kroah.com/

You are right, that is much more serious than the pretty harmless (and potentially not exploitable) patch requests sent per e-mail described in the paper. (However, the author of the commits mentioned here is not the author of the paper. If Qiushi Wu has nothing to do with all these other patches, I feel sorry for they. I still find the paper great)

Although I always try to apply Hanlon's razor and I wanted to believe that all they've done is that what's in their paper, apparently that's not the case.
This is definitely unethical, harmful and totally unacceptable.

**Comment by fr0** [comment]
> At least according to his paper he did not even sabotage the Linux Kernel.
> He has sent in the patches per e-mail and after they were approved he immediately notified the maintainers and provided them the correct fixes. None of his vulnerabilities became even just a git commit.

Don’t experiment on other humans without their knowledge or consent.

**Comment by shapowang** [comment]
吴秋实成功丢人丢到了国际上

**Comment by chrisaar** [comment]
> > At least according to his paper he did not even sabotage the Linux Kernel.
> > He has sent in the patches per e-mail and after they were approved he immediately notified the maintainers and provided them the correct fixes. None of his vulnerabilities became even just a git commit.
> 
> Don’t experiment on other humans without their knowledge or consent.

Actually, you raise an interesting point.
Was the experiment meant to target those in charge of ? Because, if so (and IANAL, just speculation) they could be in violation of their university's IRB protocols and even of the law (https://en.wikipedia.org/wiki/Human_subject_research_legislation_in_the_United_States#Rise_of_the_IRB)

**Comment by nicktimko** [comment]
> they could be in violation of their university's IRB protocols and even of the law

@chrisaar they're way ahead of you. Page 9 from the paper:
> The IRB of University of Minnesota reviewed the procedures of the experiment and determined that this is not human research. We obtained a formal IRB-exempt letter.

Even if it was dubbed "human research", I think it'd be pretty easy to get this by an IRB, especially if in their submitted protocol they said they would try as best as possible to prevent actual commits of the patches. 

Attackers won't inform you.

**Comment by sigaloid** [comment]
This did get past IRB; I think that is the main problem here. It clearly should not have been exempted. Somewhere, someone majorly messed up. Whether that be 
* the researcher misrepresenting the impact of what they are doing (a la `"oh we are just sending some modified code, to see if they complain"` vs `"we are submitting code to be added to this kernel that billions rely on, intentionally adding bugs"`) (which I think would still cause some blame to lie with IRB as they are responsible for screening this type of thing, and they should expect misrepresentation to get approval, to a certain point)
* the researcher stated exactly what the impact would be and IRB mistakenly didn't consider the consequences (in which case IRB deserves a lot more blame)

**Comment by mengdodo** [comment]
MJJ前来围观，做个人吧

**Comment by evolvingfridge** [comment]
> @sigaloid
> 
> > @ABotond https://lore.kernel.org/lkml/78ac6ee8-8e7c-bd4c-a3a7-5a90c7ccb399@roeck-us.net/
> > What does this prove?
> > This is not one of the vulnerabilities from the paper. (According to the writeup they were not even submitted from an *@umn.edu address.) Guenter Roeck even points out he doesn't see a problem with this commit (besides its limited value). UMN made actually quite a few thousand useful contributions to the kernel.
> 
> What I actually don't get around this whole scandal is that why everybody is thinking that all the other Linux contributors are trustworthy.
> The same could have been (already) done by the NSA/Chinese gov/Russian gov/etc long ago.
> 
> The paper gives a quite nice overview of the vulnerabilities introduced with small patches in the past years, how they worked and why they remained undetected for a long time.
> It helps to raise awareness of the problem. I see it no different from finding a way to get in a pretty important health care system without causing any harm then reporting it.
> OF COURSE it comes with an extra work for the maintainers - just as in case of any vulnerabilities in any systems.
> Also, I find it good if the maintainers will be more suspicious in the future about the small patches. In the middle of an ongoing cyber cold war it is more than important.

So to insure fire departments readiness to fight fires, researchers should start small fires in city to test how firefighters respond to fire, without notifying at least few people at fire department. -  your an idiot.

**Comment by janncis** [comment]
> > At least according to his paper he did not even sabotage the Linux Kernel.
> > He has sent in the patches per e-mail and after they were approved he immediately notified the maintainers and provided them the correct fixes. None of his vulnerabilities became even just a git commit.
> 
> Don’t experiment on other humans without their knowledge or consent.

Experment was on process of review not on reviewers themselfs.

**Comment by jellonek** [comment]
Can you please move this discussion to other medium as it is clearly out of the topic started by @jberkus ?

**Comment by IThinkImOKAY** [comment]
> Can you please move this discussion to other medium as it is clearly out of the topic started by @jberkus ?

no

**Comment by fr0** [comment]
> > > At least according to his paper he did not even sabotage the Linux Kernel.
> > > He has sent in the patches per e-mail and after they were approved he immediately notified the maintainers and provided them the correct fixes. None of his vulnerabilities became even just a git commit.
> > 
> > 
> > Don’t experiment on other humans without their knowledge or consent.
> 
> Experment was on process of review not on reviewers themselfs.

This is false. Ultimately, humans are the one doing the reviews, and the experiment is looking at the behavior of those humans in response to their activities. Just because they weren't named in the paper doesn't mean they weren't the target of the experiment.

**Comment by janncis** [comment]
> > > > At least according to his paper he did not even sabotage the Linux Kernel.
> > > > He has sent in the patches per e-mail and after they were approved he immediately notified the maintainers and provided them the correct fixes. None of his vulnerabilities became even just a git commit.
> > > 
> > > 
> > > Don’t experiment on other humans without their knowledge or consent.
> > 
> > 
> > Experment was on process of review not on reviewers themselfs.
> 
> This is false. Ultimately, humans are the one doing the reviews, and the experiment is looking at the behavior of those humans in response to their activities. Just because they weren't named in the paper doesn't mean they weren't the target of the experiment.

By that logic research about pc games and how players play them would need written agreement of every player, whos data was used, not just anonymised statistics, so many players got so many that kind of award. In research of review process, research does not care about who is doing review, research cares about how review is being done. Human is not a process, human is actor who follows process, at best there can be statistic about human error and some numbers on probability of error. I can do research on github about some random topic, and I will not need any permission or agreement on it, as long as it is in public domain. Bottom line, code is up for all kinds of testing, so does process on how code is added. But people got emotional about it. I loosly read paper and didn't see names, emails or references to individuals. Correct me if I'm wrong here. In fact what does this scandal signals, is that, you should not research stuff on linux because, some people got there feelings hurt.

**Comment by fr0** [comment]
> > > > > At least according to his paper he did not even sabotage the Linux Kernel.
> > > > > He has sent in the patches per e-mail and after they were approved he immediately notified the maintainers and provided them the correct fixes. None of his vulnerabilities became even just a git commit.
> > > > 
> > > > 
> > > > Don’t experiment on other humans without their knowledge or consent.
> > > 
> > > 
> > > Experment was on process of review not on reviewers themselfs.
> > 
> > 
> > This is false. Ultimately, humans are the one doing the reviews, and the experiment is looking at the behavior of those humans in response to their activities. Just because they weren't named in the paper doesn't mean they weren't the target of the experiment.
> 
> By that logic research about pc games and how players play them would need written agreement of every player, whos data was used, not just anonymised statistics, so many players got so many that kind of award. In research of review process, research does not care about who is doing review, research cares about how review is being done. Human is not a process, human is actor who follows process, at best there can be statistic about human error and some numbers on probability of error. I can do research on github about some random topic, and I will not need any permission or agreement on it, as long as it is in public domain. Bottom line, code is up for all kinds of testing, so does process on how code is added. But people got emotional about it. I loosly read paper and didn't see names, emails or references to individuals. Correct me if I'm wrong here. In fact what does this scandal signals, is that, you should not research stuff on linux because, some people got there feelings hurt.

In those examples, the subjects of the research are not directly impacted by the research. The lack of names in the paper is irrelevant.

> In fact what does this scandal signals, is that, you should not research stuff on linux because, some people got there feelings hurt.

Research on Linux is fine. Research on humans without their knowledge or consent is not, and should not be done under any circumstances. Just because you have abstracted them in your mind as being part of "the review process" does not mean they are not real and were not impacted by this study.

**Comment by AyoKeito** [comment]
> But this paper got malicious bugs into the stable branch. At what point do you stop calling it research and start calling it an attack? Is the paper enough?

It's simple, really. If they tell what malicious code they've added, and it can easily (as it should be, since it's recent) be reversed, it's a research. If those students were able to easily slide a few exploits into Linux kernel, imagine how many are there from people who make money by doing that.
But instead of acknowledging this obvious security issue, some people prefer to whine about how someone's time is wasted and proceed to just sit and wait for the next Heartbleed.
If you still think that these malicious commits made by few students are the biggest security issue Linux code has right now, read the paper again.

**Comment by janncis** [comment]
> > > > > > At least according to his paper he did not even sabotage the Linux Kernel.
> > > > > > He has sent in the patches per e-mail and after they were approved he immediately notified the maintainers and provided them the correct fixes. None of his vulnerabilities became even just a git commit.
> > > > > 
> > > > > 
> > > > > Don’t experiment on other humans without their knowledge or consent.
> > > > 
> > > > 
> > > > Experment was on process of review not on reviewers themselfs.
> > > 
> > > 
> > > This is false. Ultimately, humans are the one doing the reviews, and the experiment is looking at the behavior of those humans in response to their activities. Just because they weren't named in the paper doesn't mean they weren't the target of the experiment.
> > 
> > 
> > By that logic research about pc games and how players play them would need written agreement of every player, whos data was used, not just anonymised statistics, so many players got so many that kind of award. In research of review process, research does not care about who is doing review, research cares about how review is being done. Human is not a process, human is actor who follows process, at best there can be statistic about human error and some numbers on probability of error. I can do research on github about some random topic, and I will not need any permission or agreement on it, as long as it is in public domain. Bottom line, code is up for all kinds of testing, so does process on how code is added. But people got emotional about it. I loosly read paper and didn't see names, emails or references to individuals. Correct me if I'm wrong here. In fact what does this scandal signals, is that, you should not research stuff on linux because, some people got there feelings hurt.
> 
> In those examples, the subjects of the research are not directly impacted by the research. The lack of names in the paper is irrelevant.
> 
> > In fact what does this scandal signals, is that, you should not research stuff on linux because, some people got there feelings hurt.
> 
> Research on Linux is fine. Experimenting on humans without their knowledge or consent is not, and should not be done under any circumstances. Just because you have abstracted them in your mind as being part of "the review process" does not mean they are not real and were not impacted by this study.

Ok explain to this "dummy" who was directly impacted by study. Spent time on what seams legitimate email is fairly poor excuse for impact in research, besides asking before would invalidate entire point of this research. As far I am aware there is no one in linux that would be in position of authority to ask for it, as anyone could influence review process. You don't ask CEO for code review, but there is no one (as far I know) who would be removed from entire process. There are plenty papers that have done more and no one has bated an aye about it, because communicating it before would have invalidated purpose of research.

**Comment by fr0** [comment]
> Ok explain to this "dummy" who was directly impacted by study. Spent time on what seams legitimate email is fairly poor excuse for impact in research, besides asking before would invalidate entire point of this research.

There are multiple posts on twitter about this, such as [this one](https://twitter.com/gregkh/status/1384785747874656257) that indicate that there are real humans who are on the other side of this. I don't know all of the names of the individuals involved.

> As far I am aware there is no one in linux that would be in position of authority to ask for it, as anyone could influence review process. You don't ask CEO for code review, but there is no one (as far I know) who would be removed from entire process. There are plenty papers that have done more and no one has bated an aye about it, because communicating it before would have invalidated purpose of research.

"We can't find a way to do this study ethically" isn't a good enough reason to go forward with doing it unethically.

**Comment by janncis** [comment]
> > Ok explain to this "dummy" who was directly impacted by study. Spent time on what seams legitimate email is fairly poor excuse for impact in research, besides asking before would invalidate entire point of this research.
> 
> There are multiple posts on twitter about this, such as [this one](https://twitter.com/gregkh/status/1384785747874656257) that indicate that there are real humans who are on the other side of this. I don't know all of the names of the individuals involved.
> 
> > As far I am aware there is no one in linux that would be in position of authority to ask for it, as anyone could influence review process. You don't ask CEO for code review, but there is no one (as far I know) who would be removed from entire process. There are plenty papers that have done more and no one has bated an aye about it, because communicating it before would have invalidated purpose of research.
> 
> "We can't find a way to do this study ethically" isn't a good enough reason to go forward with doing it unethically.

First I would retrain from linking people as you are probably doing more damage than paper. 
Second they were told all patches that were bugged and provided with patches fixing there bugs and real issues. 
Third they are playing politics by removing all of university patches and thus adding old bugs. 
Fourth why on earth in few days (my guess) code got into stable release? If we go by paper, there was immediately email stating faulty patches and providing fixes for them. I have never seen any code to get into production in less than week. 
Fift as I said before there was no need to inform them of it. Hell there could be research right now about backlash, and unless there names will be named in research there will be no questions asked about if they want or not want to be part of research. 

Ps. you are not going to get informed if you get pentested in job or your actions gets reviewed for some reason. This is the same, and its open so anyone can do it. In this case it was by research, next time it can be by troll or any other entity, there is no need to ask for permission to write paper about open source review process.

**Comment by lni** [comment]
> First I would retrain from linking people as you are probably doing more damage than paper.

The best way to avoid having REAL HUMAN "linked" in such situation is to avoid having such "research" in the first place. 

> Second they were told all patches that were bugged and provided with patches fixing there bugs and real issues.

Only after those reviewers were tricked into approving their so called "incorrect minor patches". 

> Fourth why on earth in few days (my guess) code got into stable release?

They pushed the responsibility of stop merging their "incorrect minor patches" to those reviewers, don't you think that was also too selfish? 

Those are the people who spent their own time to review those "patches", those are the same people who became the human subject of the concerned "research" without their consent, according to their statement, those same people were also given an exciting new role/responsibility - to be the safe net of their "research" to prevent those "patches" to be merged. 

I simply couldn't find a more selfish way to treat those reviewers worse!

> Fift as I said before there was no need to inform them of it. Hell there could be research right now about backlash, and unless there names will be named in research there will be no questions asked about if they want or not want to be part of research.

This is according to your understand, but for the purpose of discussion let's say it is "true". The obvious problem here is those people were effectively named.

**Comment by janncis** [comment]
> > First I would retrain from linking people as you are probably doing more damage than paper.
> 
> The best way to avoid having REAL HUMAN "linked" in such situation is to avoid having such "research" in the first place, or at least obtain their consent first.
> 
> > Second they were told all patches that were bugged and provided with patches fixing there bugs and real issues.
> 
> Only after those reviewers were tricked into approving their so called "incorrect minor patches"
> 
> > Fourth why on earth in few days (my guess) code got into stable release?
> 
> They pushed the responsibility of stop merging their "incorrect minor patches" to those reviewers, don't you think that was selfish as well?
> 
> Those are the people who spent their own time to review those "patches", those are the same people who became the human subject of the concerned "research" without their consent, according to their statement, those same people were also given an exciting new role/responsibility - to be the safe net of their "research" to prevent those "patches" to be merged.
> 
> I simply couldn't find a more selfish way to treat those reviewers worse!
> 
> > Fift as I said before there was no need to inform them of it. Hell there could be research right now about backlash, and unless there names will be named in research there will be no questions asked about if they want or not want to be part of research.
> 
> This is according to your understanding, but for the purpose of discussion let's say it is "true". The obvious problem here is those people were effectively named.

Yes lets ask for every research consent and then wonder why some research data is bad and unusable. As far I know, from loosely reading paper, no names where given so people them selfs put there names out or as that guy, named them. To be honest I would be more pissed at that guy for naming me than for being subject in review research, but that's just me. 

Well they are safety net of linux, and yes lets as for consent in research where asking for consent would invalidate collected data. Where is logic there?

Ps. If im wrong on paper not naming people please correct me with facts from paper, and i will apologise for it.

**Comment by evolvingfridge** [comment]
> > > First I would retrain from linking people as you are probably doing more damage than paper.
> > 
> > 
> > The best way to avoid having REAL HUMAN "linked" in such situation is to avoid having such "research" in the first place, or at least obtain their consent first.
> > > Second they were told all patches that were bugged and provided with patches fixing there bugs and real issues.
> > 
> > 
> > Only after those reviewers were tricked into approving their so called "incorrect minor patches"
> > > Fourth why on earth in few days (my guess) code got into stable release?
> > 
> > 
> > They pushed the responsibility of stop merging their "incorrect minor patches" to those reviewers, don't you think that was selfish as well?
> > Those are the people who spent their own time to review those "patches", those are the same people who became the human subject of the concerned "research" without their consent, according to their statement, those same people were also given an exciting new role/responsibility - to be the safe net of their "research" to prevent those "patches" to be merged.
> > I simply couldn't find a more selfish way to treat those reviewers worse!
> > > Fift as I said before there was no need to inform them of it. Hell there could be research right now about backlash, and unless there names will be named in research there will be no questions asked about if they want or not want to be part of research.
> > 
> > 
> > This is according to your understanding, but for the purpose of discussion let's say it is "true". The obvious problem here is those people were effectively named.
> 
> Yes lets ask for every research consent and then wonder why some research data is bad and unusable. As far I know, from loosely reading paper, no names where given so people them selfs put there names out or as that guy, named them. To be honest I would be more pissed at that guy for naming me than for being subject in review research, but that's just me.
> 
> Well they are safety net of linux, and yes lets as for consent in research where asking for consent would invalidate collected data. Where is logic there?
> 
> Ps. If im wrong on paper not naming people please correct me with facts from paper, and i will apologise for it.

Your ignorant attitude is just disgusting, consent can be structured in many ways, there's no need to invent any new research methods to a very large degree, for example interested volunteers can only give consent for participation without knowing any details (including when it will happen) of study. Furthermore, what researcher(s) did is red-teaming, without authorization(s) and designated supervisor(s) from organization itself and beyond this is illegal behavior that will result in jail time.

**Comment by janncis** [comment]
> > Yes lets ask for every research consent and then wonder why some research data is bad and unusable. As far I know, from loosely reading paper, no names where given so people them selfs put there names out or as that guy, named them. To be honest I would be more pissed at that guy for naming me than for being subject in review research, but that's just me.
> > Well they are safety net of linux, and yes lets as for consent in research where asking for consent would invalidate collected data. Where is logic there?
> > Ps. If im wrong on paper not naming people please correct me with facts from paper, and i will apologise for it.
> 
> Your ignorant attitude is just disgusting, consent can be structured in many ways, there's no need to invent any new research methods to a very large degree, for example interested volunteers can only give consent for participation without knowing any details (including when it will happen) of study. Furthermore, what researcher(s) did is red-teaming, without authorization(s) and designated supervisor(s) from organization itself and beyond this is illegal behavior that will result in jail time.

Well I don't really care about it, if you find it disgusting or funny, I find this "cancel culture" of everything that offends someone disgusting, so what? Should I cry now?

I think you have no idea what you are talking about, it might have been red-teaming, but since its open source its not illegal to do so, besides they only sent code via email and it was vetted by linux own staff, you will be hard pressed to find judge who will think the same. open source == can test the hell out of it, if you want it, including the way new code is added. 
Deamit its a code not soft, you cant easly vet it, and the small one at that. 

But no people like to get butt hurt over, just because there religion to oss was painted in a bit darker tone.

**Comment by evolvingfridge** [comment]
> > > Yes lets ask for every research consent and then wonder why some research data is bad and unusable. As far I know, from loosely reading paper, no names where given so people them selfs put there names out or as that guy, named them. To be honest I would be more pissed at that guy for naming me than for being subject in review research, but that's just me.
> > > Well they are safety net of linux, and yes lets as for consent in research where asking for consent would invalidate collected data. Where is logic there?
> > > Ps. If im wrong on paper not naming people please correct me with facts from paper, and i will apologise for it.
> > 
> > 
> > Your ignorant attitude is just disgusting, consent can be structured in many ways, there's no need to invent any new research methods to a very large degree, for example interested volunteers can only give consent for participation without knowing any details (including when it will happen) of study. Furthermore, what researcher(s) did is red-teaming, without authorization(s) and designated supervisor(s) from organization itself and beyond this is illegal behavior that will result in jail time.
> 
> Well I don't really care about it, if you find it disgusting or funny, I find this "cancel culture" of everything that offends someone disgusting, so what? Should I cry now?
> 
> I think you have no idea what you are talking about, it might have been red-teaming, but since its open source its not illegal to do so, besides they only sent code via email and it was vetted by linux own staff, you will be hard pressed to find judge who will think the same. open source == can test the hell out of it, if you want it, including the way new code is added.
> Deamit its a code not soft, you cant easly vet it, and the small one at that.
> 
> But no people like to get butt hurt over, just because there religion to oss was painted in a bit darker tone.

go way, back to your qanon social media.

**Comment by IThinkImOKAY** [comment]
@janncis If you'd like a place where you can talk to others without brains, reddit is down the hall and to the left ;)
